folly/container/detail/F14Table.h

```diff
@@ -187,6 +187,16 @@ struct StdNodeReplica<
 
 #endif
 
+template <class Container, class Predicate>
+void erase_if_impl(Container& c, Predicate predicate) {
```

Comment:
While it follows the std for `erase_if` to take `Predicate` by value, the trouble comes when `erase_if` calls `erase_if_impl` and attempts to copy `Predicate` - the type may not be copy-constructible. We can take `Predicate&` here.

Reply:
Right, I followed the std in API here, but I'm fine taking `Predicate&`.

The predicate could now be a `folly::Function` which isn't copyable, which may be a real use case. In practice, I would expect most predicates here to be cheaply copyable lambdas.